### PR TITLE
Adding more cloud nodes

### PR DESCRIPTION
**Pull request by cskiraly**
to test, use docker image `agileiot/agile-nodered-armv7l:cloud-nodes`

@Exulansis please verify why the solid node is not showing up.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -26,6 +26,6 @@
 [submodule "node-red-contrib-agile-solid"]
 	path = node-red-contrib-agile-solid
 	url = https://github.com/Agile-IoT/node-red-contrib-agile-solid.git
-[submodule "node-red-contrib-thingspeak42"]
-	path = node-red-contrib-thingspeak42
-	url = https://github.com/clough42/node-red-contrib-thingspeak42.git
+[submodule "node-red-contrib-agile-thingspeak"]
```
Comment (rosogon) [inline .gitmodules]: Why this change? We don't have this repository yet. I don't know if @Exulansis is planning to fork the thingspeak42 node, but let's change it when forked.

Comment (cskiraly) [reply .gitmodules]: It is actually there. I was also surprised to see it, since GitHub search is not showing it. But then, browsing through the repose from the GitHub repo list, I've found it. 
I suppose their search is going from off-line data build each night, or something similar. Or maybe they avoid showing forks without modifications.

Comment (cskiraly) [reply .gitmodules]: @Exulansis  , what do you think, which one should we keep?

Comment (rosogon) [reply .gitmodules]: It is true! I have forgotten it. @Exulansis made some minor modifications, so I think it is right to keep the agile repo.

File: Dockerfile [hunk 2]
```diff
@@ -19,16 +19,35 @@ WORKDIR node-red-contrib-security-nodes
...
+COPY $NODE $NODE
+RUN npm install $NODE
+
+ARG NODE=node-red-contrib-agile-googledrive
+COPY $NODE $NODE
+RUN npm install $NODE
+
+ARG NODE=node-red-contrib-agile-thingspeak
+COPY $NODE $NODE
+RUN npm install $NODE
```
Comment (rosogon) [inline Dockerfile]: This Dockerfile only installs agile-node-red-nodes and node-red-contrib-security-nodes. I will try to check why.

File: Dockerfile [hunk 3]
```diff
@@ -19,16 +19,35 @@ WORKDIR node-red-contrib-security-nodes
...
-RUN npm install agile-node-red-nodes
+ARG NODE=agile-node-red-nodes
+COPY $NODE $NODE
+RUN npm install $NODE
 
 # adding Xively support
-COPY node-red-contrib-agile-xively node-red-contrib-agile-xively
-RUN npm install node-red-contrib-agile-xively
+ARG NODE=node-red-contrib-agile-xively 
+COPY $NODE $NODE
```
Comment (rosogon) [inline Dockerfile]: It looks this COPY does not have effect in the container. I enter the container and node-red-contrib-agile-xively is not there. Could it be that only the first ARG NODE=... is considered?

**Comment by cskiraly** [comment]
What version of docker (not CLI, but server) are you running?

**Comment by rosogon** [comment]
Docker version 1.12.3, build 6b644ec

Does it work for you?

**Comment by rosogon** [comment]
What comes in resin.io distribution:
Docker version 1.10.3, build 20f81dd-unsupported

Does it work for you? Repeating ARG variables looks strange to the expected usage of ARG.

**Comment by cskiraly** [comment]
Well, behavior with redefinition of ARG is not specified at all.  However, ARG is defined in relation to ENV, where it is (should be) supported.
To clarify, I will check again the ARG version, but independent of the results, we can use your fix. Thanks!

**Comment by rosogon** [comment]
SOLID does not appear in the palette, but it is an issue of the node itself: it does not declare the node-red section inside package.son.

**Review by rosogon — APPROVED** [review]

**Comment by cskiraly** [comment]
I've just checked with Docker version 17.05.0-ce, build 89658be. Redefinition of ARG work as expected (by me :-) ). Anyway, we keep your version for increased backward compatibility.

**Comment by cskiraly** [comment]
Rebased on top of actual master.
